src/components/layer-controller/LayerController.js

```diff
@@ -54,39 +69,51 @@ export default function LayerController({ imageData, layerId, handleLayerRemove
   const [colormap, setColormap] = useState(DEFAULT_LAYER_PROPS.colormap);
   const [opacity, setOpacity] = useState(DEFAULT_LAYER_PROPS.opacity);
   const [channels, dispatch] = useReducer(reducer, {});
-
-  /*
-  * TODO: UI selectors for channels just assume the first dimension (so we only support
-  * simple 2D images, with one additonal dimension (i.e. mz, time, channel, z).
-  * We will need to come up with more than just a single drop down for selecting image panes
-  * from multi-dimensional images.
-  */
-  const [dimName, channelOptions, defaultSelection] = useMemo(() => {
-    const { metadata: { dimensions } } = imageData;
-    const { values, field } = dimensions[0];
-    return [field, values, { [field]: 0 }];
-  }, [imageData]);
+  const [dimensions, setDimensions] = useState([]);
 
   useEffect(() => {
     initLoader(imageData).then((loader) => {
+      const loaderDimensions = loader.dimensions;
+      setDimensions(loaderDimensions);
       PubSub.publish(LAYER_ADD, {
         layerId,
         loader,
         layerProps: DEFAULT_LAYER_PROPS,
       });
-      // Add channel on image add automatically
+      // Add channel on image add automatically as the first avaialable value for each dimension.
+      const defaultSelection = Object.assign(
+        {},
+        ...loaderDimensions.map(
+          dimension => dimension.type !== 'quantitative' && {
+            [dimension.field]: dimension.values[0],
+          },
+        ),
+      );
       dispatch({
         type: 'ADD_CHANNEL',
         layerId,
         payload: { selection: defaultSelection },
       });
     });
-  }, [layerId, imageData, defaultSelection]);
+  }, [layerId, imageData]);
 
   const handleChannelAdd = () => dispatch({
     type: 'ADD_CHANNEL',
     layerId,
-    payload: { selection: defaultSelection },
+    payload: {
+      selection: Object.assign(
+        {},
+        ...dimensions.map(
+          // Set new image to default selection for non-global channels
+          // and use current global selection otherwise.
+          dimension => dimension.type !== 'quantitative' && {
+            [dimension.field]: GLOBAL_SLIDER_DIMENSION_FIELDS.indexOf(dimension.field) >= 0
+              ? channels[Object.keys(channels)[0]].selection[dimension.field]
+              : dimension.values[0],
+          },
+        ),
+      ),
```

Comment:
Having a hard time understanding what this code does. Shouldn't this be the same as the image layer add default? That's why we had react memo before...

Reply:
With global selections, the `defaultSelection` needs to be able to change now as the global selection changes.  Hence, we need to use the default selection for non-globally controlled selectors (in this case channel), and override with the current global selection.